lab_4_fill_words_by_ngrams/main.py

```diff
@@ -98,6 +127,32 @@ def run(self, seq_len: int, prompt: str) -> str:  # type: ignore
                 or if sequence has inappropriate length,
                 or if methods used return None.
         """
+        if not (isinstance(seq_len, int) and isinstance(prompt, str) and seq_len > 0 and prompt):
+            raise ValueError
+        encoded = self._word_processor.encode(prompt)
+        if not encoded:
+            raise ValueError
+        current_seq = 0
+        while seq_len > current_seq:
```

Comment:
Can you explain why you preferred while over for here?

Reply:
I thought it was more logical here because as I understood we don't have fixed sequence and I thought that loop'd continue generating tokens until the needed length is reached